### DISCUSSION TITLE
Motion estimate

### DISCUSSION THREAD

**Discussion by araikes** [dipy/dipy#2874] [3. Community Lounge]
I'm using motion correction here in DIPY for my rodent data and I wanted to compute some metric of motion (like FD/DVARS). 

Admittedly, I'm just not smart enough for this part... is there a way to take the affines from the motion correction and get translation and rotation components akin e.g. https://tortoise.nibib.nih.gov/community/can-tortoise-report-head-motion-index

Thanks

**Comment by skoudoro** [comment]
After the motion correction you should get the transformation affine (need to check if it returns the affine, if not, this is a feature to add)
- the translation components are `self.affine[:-1, -1:]`
- the rotation components are `self.affine[:-1, :-1]`

there is also [this function](https://github.com/dipy/dipy/blob/fa699a95bb4e7a7bc33a66333b14413171ecd8e9/dipy/align/tests/test_imaffine.py#L358) that can help you to understand. Does it answer your question?

ps converting to discussion since it is not an issue.

**Comment by araikes** [comment]
Thanks @skoudoro. That answers it.

**Comment by arokem** [comment]
I don't think that's accurate, because the affines also contain scaling, shear that mix in with the rotation component. 

This skimage object could be useful to decompose an affine into these components: https://scikit-image.org/docs/stable/api/skimage.transform.html#skimage.transform.AffineTransform

**Reply by Garyfallidis** [reply]
I believe we have a way to decompose that affine matrix and get back the motion parameters. Correct @skoudoro and @arokem ?

**Reply by Garyfallidis** [reply]
Are we forgetting our own API? 
@araikes see this please https://github.com/dipy/dipy/blob/master/dipy/align/streamlinear.py#L1437

**Reply by arokem** [reply]
Yes, we are! `decompose_matrix` should do it!

**Reply by Garyfallidis** [reply]
@araikes please use the decompose function and provide feedback. @skoudoro we were moving all sorts of similar functionality to the transforms submodule. Is this ongoing?

**Reply by araikes** [reply]
Hah! that'd do it... I'll run it and let you know.

I too forgot that was in there.
